hazelcast/src/hazelcast/client/xml_impl.cpp

```diff
@@ -0,0 +1,1440 @@
+  const std::string& config_file,
+  std::vector<std::string> accepted_suffixes)
+{
+    std::string config_file_lower;
+    if (config_file.empty()) {
+        config_file_lower = config_file;
+    } else {
+        config_file_lower = config_file;
+        boost::algorithm::to_lower(config_file_lower);
+    }
+    int last_dot_index = (int)config_file_lower.find_last_of('.');
+    if (last_dot_index == -1) {
+        return false;
+    }
+    std::string config_file_suffix =
+      config_file_lower.substr(last_dot_index + 1);
+    return std::find(accepted_suffixes.begin(),
+                     accepted_suffixes.end(),
+                     config_file_suffix) != accepted_suffixes.end();
+}
+
+std::ifstream*
+abstract_config_locator::get_in()
+{
+    return in;
+}
+
+bool
+abstract_config_locator::load_from_working_directory(
+  const std::string& config_file_path)
+{
+    try {
+        std::FILE* file;
+        file = fopen(config_file_path.c_str(), "r");
+        if (file == nullptr) {
+            std::cout << "FINEST: "
+                      << "Could not find " + config_file_path +
+                           " in the working directory."
+                      << std::endl;
+            return false;
+        }
+        std::cout << "INFO: Loading " + config_file_path +
+                       " from the working directory."
+                  << std::endl;
+        configuration_file = file;
+        in = new std::ifstream();
+        in->open(config_file_path, std::ios::in);
+        if (in->fail()) {
+            throw hazelcast::client::exception::hazelcast_(
+              "Failed to open file: " + config_file_path);
+        }
+        return true;
+    } catch (const std::runtime_error& e) {
+        throw hazelcast::client::exception::hazelcast_(e.what());
+    }
+}
+
+bool
+abstract_config_locator::load_from_working_directory(
+  const std::string& config_file_prefix,
+  const std::vector<std::string>& accepted_suffixes)
+{
+    if (accepted_suffixes.empty()) {
+        throw std::invalid_argument(
+          "Parameter acceptedSuffixes must not be empty");
+    }
+    for (const auto& suffix : accepted_suffixes) {
+        if (suffix.empty()) {
+            throw std::invalid_argument(
+              "Parameter acceptedSuffixes must not contain empty strings");
+        }
+        if (load_from_working_directory(config_file_prefix + "." + suffix)) {
+            return true;
+        }
+    }
+    return false;
+}
+bool
+abstract_config_locator::load_from_system_property(
+  const std::string& property_key,
+  const std::vector<std::string>& accepted_suffixes)
+{
+    return load_from_system_property(property_key, false, accepted_suffixes);
+}
+void
+abstract_config_locator::load_system_property_file_resource(
+  const std::string& config_system_property)
+{
+    configuration_file = fopen(config_system_property.c_str(), "r");
+    std::cout << "Using configuration file at " << config_system_property
```

Comment:
java writes absolute file path in the log. You can get absolute path using [boost filesystem API](https://www.thecodeteacher.com/question/213860/How-get-file-path-from-std::ifstream-in-c++).